### PR TITLE
Remove outdated docs about pytest.warns and DeprecatedWarning

### PR DESCRIPTION
**Pull request by nicoddemus**
Since #2908, the user doesn't need to set warning filters to capture
`DeprecationWarning` with `pytest.warns`.

Fix #8666

### CODE REVIEW

File: testing/test_warnings.py [hunk 1]
```diff
@@ -516,6 +516,24 @@ def test_hidden_by_system(self, pytester: Pytester, monkeypatch) -> None:
         result = pytester.runpytest_subprocess()
         assert WARNINGS_SUMMARY_HEADER not in result.stdout.str()
 
+    def test_recwarn(self, pytester: Pytester, monkeypatch) -> None:
+        """
+        Check that recwarn can capture DeprecationWarning by default
+        without custom filterwarnings (see #8666).
+        """
+        pytester.makepyfile(
```
Comment (asottile) [inline testing/test_warnings.py:524]: I think this test should be inside test_recwarn.py?

also I don't think `pytester` is needed here -- this can be tested directly (?)

the testsuite is failing because this test function name matches `test_recwarn.py`

Comment (nicoddemus) [reply testing/test_warnings.py]: Good points, all done.

I initially used pytester to ensure our own warning filters wouldn't interfere with the test, but in the end I'm not sure it is needed after all.

File: src/_pytest/tmpdir.py [hunk 2]
```diff
@@ -131,8 +131,8 @@ def getbasetemp(self) -> Path:
             # rootdir's permissions. Historically 0o755 was used, so we can't
             # just error out on this, at least for a while.
             if hasattr(os, "getuid"):
+                uid = os.getuid()  # type:ignore[attr-defined]
```
Comment (nicoddemus) [inline src/_pytest/tmpdir.py:134]: This was done in a separate commit (84c5fe4d440faa70b69e618779828af55436b569).

Comment (asottile) [reply src/_pytest/tmpdir.py:134]: I think the `hasattr` check should be `if sys.platform != 'win32'`  -- this matches the guard from typeshed: https://github.com/python/typeshed/blob/48a346920bbc24165bd9e9cf29a2b5140eae2abc/stdlib/os/__init__.pyi#L388

**Review by asottile — APPROVED** [review]

File: src/_pytest/tmpdir.py (hunk 2 above)
Comment (nicoddemus) [reply src/_pytest/tmpdir.py]: Agree, thanks. Updated. 👍

**Comment by nicoddemus** [comment]
Backport: https://github.com/pytest-dev/pytest/pull/8755
